### DISCUSSION TITLE
What can cause reset of HKLM\SOFTWARE\Microsoft\Windows\CurrentVersion\Lxss

### DISCUSSION THREAD

**Discussion by szarykott** [microsoft/WSL#11685] [Q&A]
I've been trying to change subnets used by WSL2 in order to make it work with VPN. 

As mentioned in many solutions, I edited `HKLM\SOFTWARE\Microsoft\Windows\CurrentVersion\Lxss` registry entry to desired values.

However, each time I start WSL after reboot or `wsl --shutdown` values are reset to default. I am puzzled. Did anyone encounter something like it?

**Comment by jhe-pf** [comment]
I did encounter the same, just now.

After playing around, asking the internet and Github Copilot:
The values always get overwritten when the wsl is shutting down (either by manually shutting it down or rebooting), as you experienced yourself.

The trick is: 
1. Shutdown the wsl
2. Change the values
3. Do an immediate reboot

At the next boot, when wsl gets started, it reads the new/your values from the registry and from that moment in time they are used by the wsl. (and of course: written back when shutting down the wsl)
